lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -8346,7 +8346,19 @@ void EmitMemberNode(ParseNode *memberNode, Js::RegSlot objectLocation, ByteCodeG
         // Transparently pass the name expr
         // The Emit will replace this with a temp register if necessary to preserve the value.
         nameNode->location = nameNode->AsParseNodeUni()->pnode1->location;
+
+        // Save the previous value of the flag to be restored later.
+        bool prevFlag = byteCodeGenerator->forceStrictModeForClassComputedPropertyName;
+
+        // Strict mode must be enforced on the evaluation of computed property names inside  
+        // classes, thus enable the flag if the computed property name is a class member.
+        byteCodeGenerator->forceStrictModeForClassComputedPropertyName = isClassMember;
```

Comment:
I feel like maybe here we should be setting `isClassMember || prevFlag`. Consider the following (which might be a nice test case too): `class C { [{[a.b=3]: 4}[3]]() {} }`. Inside the computed method name on the class (strict mode), we have a computed property name for an object literal (which I believe should also be strict mode, because strict mode affects anything lexically inside the expression).